### DISCUSSION TITLE
[2023-09-04] GUI Team Demo Day

### DISCUSSION THREAD

**Discussion by wdanilo** [enso-org/enso#7722] [Weekly Demo Day]
I was working on the new website. Not much visible yet, but we have all content written down and basic template ready :) : 

https://github.com/enso-org/enso/assets/1623053/639286f3-4e92-4215-a4a9-538398a428a7

**Comment by farmaazon** [comment]
A lot of discussions related to migrating IDE to Vue framework. Also learned a thing or two about it and I was able to develop a view of the component list in the browser with selection highlight as in the design; you can navigate with arrows, things are selected on mouse hover, and - as a bonus - home and end keys also work (I always missed them in rust IDE).

https://github.com/enso-org/enso/assets/3919101/3c054d63-540c-4532-a5f9-31811c568254

In this implementation, only the visible entries are added to HTML. We do _not_ re-use the existing HTML-s, but I deemed it's not necessary - we already can fast-scroll 300 entries, as in the video.

**Comment by vitvakatu** [comment]
This week mainly working on https://github.com/enso-org/enso/issues/6417

The implementation is done, but has a few bugs, so I'm finishing it today.


https://github.com/enso-org/enso/assets/6566674/f9f8086c-13ff-42e5-a1f5-bb2d13b74954

**Comment by Frizi** [comment]
This week I was introducing Vue framework to GUI team members, and preparing base architecture of the new gui app. Moved the graph data model to Yjs data structures, allowing synchronization between clients editing the same project.


https://github.com/enso-org/enso/assets/919491/5e299e17-a55d-41f8-af57-9379cf2f4a4b

**Comment by JaroslavTulach** [comment]
_Engine & Language Server_ work:
- Faster & more reliable logging: #7559
- `Any.to` type conversion method: #7704
- Suggestion database fixes: #7699
- Robustness fixes: #7668 & #7659
- Apache Arrow likes Python `pandas` #7678
